pkg/sources/reconciler/awssnssource/subscription.go

```diff
@@ -129,17 +129,8 @@ func (r *Reconciler) ensureUnsubscribed(ctx context.Context) error {
 
 	src := v1alpha1.SourceFromContext(ctx)
 
-	sm := src.GetStatusManager()
-
-	// Note: requiring readiness to unsubscribe might leak subscriptions when the service
-	// is not ready. The reason for requiring IsReady is because we require the status address
-	// when looking for the SNS subscription and it is informed when the service is ready.
-	if !sm.IsReady() {
-		event.Warn(ctx, ReasonFailedUnsubscribe, "SNS status information incomplete, skipping finalization")
-		return nil
-	}
```

Comment:
This code (which I wrote) was wrong.

A Knative Service will in most cases have a URL even if it's not ready, so this check is not only unnecessary, but can also lead to leaking SNS subscriptions, which this PR tries to prevent.